lib/trip_dispatcher.rb

```diff
@@ -27,14 +35,51 @@ def inspect
               #{passengers.count} passengers>"
     end
 
+    def find_available_drivers
+      available_drivers = []
+      @drivers.each do |driver|
+        if driver.status == :AVAILABLE
+          available_drivers << driver
+        end
+      end
+      return available_drivers
+    end
+
+    # Wave 3: creates a trip and assigns an available driver
+    def request_trip(passenger_id)
+      dispatch_driver = find_available_drivers
+      if dispatch_driver.length == 0
+        raise ArgumentError.new("No available drivers")
+      end
+      trip = Trip.new(
+        id: @trips.last.id + 1,
+        passenger: find_passenger(passenger_id),
+        passenger_id: passenger_id,
+        start_time: Time.now,
+        end_time: nil,
+        cost: nil,
+        rating: nil,
+        driver: dispatch_driver[0],
+        driver_id: dispatch_driver[0].id,
+      )
+      dispatch_driver[0].add_trip(trip)
+      dispatch_driver[0].status = :UNAVAILABLE
+      trip.passenger.add_trip(trip)
+      @trips << trip
+      trip.connect(trip.passenger)
+      trip.connect_driver(dispatch_driver[0])
```

Comment:
Similarly, on line 67, you're calling the `add_trip` method on the passenger,

but in line 69, you call `connect` on trip, which also calls the `add_trip` methd on passenger.

How do you need to refactor this logic so that we're only adding the trip to the passenger once? Where does it make the most sense to put this logic-- on `TripDispatcher`, `Trip`, or `Passenger`? What information do you need?

Lastly, what tests would you all have needed to write in order to catch this bug?